### PR TITLE
Add xxhash template function

### PR DESCRIPTION
**Pull request by oriubimo**

### CODE REVIEW

**Comment by oriubimo** [comment]
The xxhash variadic API is easier to use than having to call the hash function for every member, that's why I added this.

**Comment by romange** [comment]
I will look at in at weekend.

File: base/hash.h [hunk 1]
```diff
@@ -33,7 +38,98 @@ inline uint32_t Fingerprint32(const std::string& str) {
   return uint32_t(res >> 32) ^ uint32_t(res);
 }
 
+namespace detail {
+
+// Should use std::has_unique_object_representations but we do not have in C++14.
+template<typename Hasher, typename T>
+std::enable_if_t<std::is_arithmetic<T>::value || std::is_enum<T>::value>
```
Comment (romange) [inline base/hash.h]: Pls restrict to is_integral, handling double types correctly would require more work.

File: base/hash.h [hunk 2]
```diff
@@ -33,7 +38,98 @@ inline uint32_t Fingerprint32(const std::string& str) {
...
+  void update(void const* key, std::size_t len) noexcept {
+    XXH64_update(&state_, key, len);
+  }
+
+  unsigned long long digest() noexcept {
+    return XXH64_digest(&state_);
+  }
+};
+
+} //namespace detail
```
Comment (romange) [inline base/hash.h]: <space><space>//<space>namespace detail

**Comment by oriubimo** [comment]
I have applied the requested fixes, waiting for your confirmation.

**Pull request merged**